dali/pipeline/data/tensor_list.h

```diff
@@ -265,7 +270,7 @@ class DLL_PUBLIC TensorList {
     this->set_order(order);
 
     // Save our new pointer and bytes. Reset our type, shape, and size
-    data_.set_backing_allocation(ptr, bytes, pinned, type, shape.num_elements());
+    data_.set_backing_allocation(ptr, bytes, pinned, type, shape.num_elements(), device_id, order);
```

Comment:
So now we set the order twice, right? Do we care, should the `set_backing_allocation` do it alone?

Reply:
Good catch, removing

Reply:
On the second thought, I kept it and expanded the comment, it will be a no op in set_backing_allocation, but maybe it's worth to keep things explicit.